fix: SmrtObject system properties causing circular serialization errors

## Issue Description

When saving SMRT objects that extend SmrtObject, the serializer attempts to serialize system properties (`ai`, `db`, `fs`, `system_db`, `signal_bus`) which causes errors when these properties contain objects with circular references (like the OpenAI client).

## Error

```
Caused by: TypeError: Converting circular structure to JSON
    --> starting at object with constructor 'OpenAI'
    |     property 'completions' -> object with constructor 'Completions'
    --- property '_client' closes the circle
 ❯ serializeValue ../sdk/packages/sql/src/sqlite.ts:252:21
 ❯ serializeRecord ../sdk/packages/sql/src/sqlite.ts:269:27
 ❯ Object.upsert ../sdk/packages/sql/src/sqlite.ts:434:30
```

## Current Behavior

The generated schema includes these system properties:
```sql
CREATE TABLE IF NOT EXISTS "councils" (
  ...
  "system_db" REAL,
  "fs" REAL,
  "db" REAL,
  "ai" REAL,
  "signal_bus" REAL
);
```

When `council.save()` is called, it tries to serialize the `ai` property which contains the OpenAI client.

## Expected Behavior

System properties should either be:
1. Marked as transient and excluded from serialization
2. Not included in the database schema at all
3. Handled specially during serialization to avoid circular reference errors

## Steps to Reproduce

1. Create a SMRT object that extends SmrtObject
2. Initialize with AI client
3. Call `.save()` on the object
4. Observe circular structure serialization error

## Context

Discovered while running praeco tests after updating to latest SDK/SMRT versions (SDK 0.47.1, SMRT with fix #72/66).

Blocking praeco tests from passing. As per development workflow, praeco development is paused until this issue is resolved.

## Related Files

- `praeco/src/council.ts` - Example SMRT object
- `sdk/packages/sql/src/sqlite.ts` - Serialization logic
- `smrt/packages/core/src/object.ts` - SmrtObject base class

## Related Projects

- happyvertical/praeco - Currently blocked by this issue

--- a/packages/core/src/__tests__/issue-75-circular-serialization.test.ts
+++ b/packages/core/src/__tests__/issue-75-circular-serialization.test.ts
@@ -1,0 +1,149 @@
+/**
+ * Test for Issue #75: Circular Serialization Errors
+ *
+ * This test verifies that:
+ * 1. System properties from base classes are excluded from schema
+ * 2. Only user-defined properties and whitelisted schema properties are included
+ * 3. Serialization does not cause circular reference errors
+ */
+
+import { writeFileSync } from 'node:fs';
+import { tmpdir } from 'node:os';
+import { join } from 'node:path';
+import { describe, expect, it } from 'vitest';
+import { SmrtObject } from '../object';
+import { ASTScanner } from '../scanner/ast-scanner';
+
+describe('Issue #75: Circular Serialization Errors', () => {
+  describe('AST Scanner excludes base class properties', () => {
+    it('should exclude system properties from SmrtClass and SmrtObject', () => {
+      // Note: This test scans actual source files including base classes
+      // The AST scanner will extract base class properties and exclude them from subclass schemas
+
+      // Scan the actual object.ts file which contains SmrtObject base class
+      const objectFilePath = join(process.cwd(), 'src', 'object.ts');
+      const scanner = new ASTScanner([objectFilePath]);
+      const results = scanner.scanFiles();
+
+      // SmrtObject itself should not be included (it's a base class)
+      const smrtObjectDef = results[0]?.objects.find(
+        (obj) => obj.className === 'SmrtObject',
+      );
+      expect(smrtObjectDef).toBeUndefined();
+
+      // When we scan actual SMRT objects in tests below,
+      // they register at runtime and we can verify system properties are excluded
+    });
+  });
+
+  describe('Serialization without circular references', () => {
+    it('should serialize object to JSON without circular reference errors', async () => {
+      class TestProduct extends SmrtObject {
+        productName: string = 'Test Product';
+        price: number = 99.99;
+        inStock: boolean = true;
+      }
+
+      // Create instance with minimal options
+      const product = new TestProduct({
+        name: 'Widget',
+        db: ':memory:',
+      });
+
+      await product.initialize();
+
+      // This should not throw circular reference errors
+      let jsonString: string;
+      expect(() => {
+        jsonString = JSON.stringify(product);
+      }).not.toThrow();
+
+      // Verify the JSON contains user properties
+      const parsed = JSON.parse(jsonString!);
+      expect(parsed).toHaveProperty('productName', 'Test Product');
+      expect(parsed).toHaveProperty('price', 99.99);
+      expect(parsed).toHaveProperty('inStock', true);
+      expect(parsed).toHaveProperty('name', 'Widget');
+
+      // Verify system properties are not serialized
+      expect(parsed).not.toHaveProperty('_ai');
+      expect(parsed).not.toHaveProperty('_db');
+      expect(parsed).not.toHaveProperty('_fs');
+      expect(parsed).not.toHaveProperty('ai');
+      expect(parsed).not.toHaveProperty('db');
+      expect(parsed).not.toHaveProperty('fs');
+    });
+
+    it('should save object without circular reference errors', async () => {
+      class TestArticle extends SmrtObject {
+        title: string = 'Test Article';
+        body: string = 'Article content';
+        published: boolean = false;
+      }
+
+      const article = new TestArticle({
+        name: 'My Article',
+        db: ':memory:',
+      });
+
+      await article.initialize();
+
+      // This should not throw circular reference errors during save
+      await expect(article.save()).resolves.not.toThrow();
+
+      // Verify we can create a new instance and load it
+      // Important: Use the same database instance (article.db) for retrieval
+      const retrieved = new TestArticle({
+        id: article.id,
+        db: article.db, // Reuse the same database instance
+      });
+      await retrieved.initialize();
+      await retrieved.loadFromId();
+
+      expect(retrieved.title).toBe('Test Article');
+      expect(retrieved.body).toBe('Article content');
+    });
+  });
+
+  describe('toJSON() method behavior', () => {
+    it('should only include schema fields in toJSON() output', async () => {
+      class TestBook extends SmrtObject {
+        bookTitle: string = 'The Book';
+        author: string = 'Author Name';
+        pages: number = 300;
+      }
+
+      const book = new TestBook({
+        name: 'Book 1',
+        db: ':memory:',
+      });
+
+      await book.initialize();
+
+      const json = book.toJSON();
+
+      // User-defined fields should be present
+      expect(json).toHaveProperty('bookTitle', 'The Book');
+      expect(json).toHaveProperty('author', 'Author Name');
+      expect(json).toHaveProperty('pages', 300);
+
+      // Schema properties should be present
+      expect(json).toHaveProperty('id');
+      expect(json).toHaveProperty('slug');
+      expect(json).toHaveProperty('context');
+      expect(json).toHaveProperty('name', 'Book 1');
+      expect(json).toHaveProperty('created_at');
+      expect(json).toHaveProperty('updated_at');
+
+      // System properties should NOT be present
+      expect(json).not.toHaveProperty('_ai');
+      expect(json).not.toHaveProperty('_db');
+      expect(json).not.toHaveProperty('_fs');
+      expect(json).not.toHaveProperty('_className');
+      expect(json).not.toHaveProperty('options');
+      expect(json).not.toHaveProperty('db');
+      expect(json).not.toHaveProperty('ai');
+      expect(json).not.toHaveProperty('fs');
+    });
+  });
+});

--- a/packages/core/src/registry.ts
+++ b/packages/core/src/registry.ts
@@ -293,10 +293,47 @@ export class ObjectRegistry {
           currentProto = Object.getPrototypeOf(currentProto);
         }
 
+        // Properties to exclude from schema (system/internal properties from base classes)
+        // These cause circular serialization errors when included
+        const BASE_CLASS_PROPERTIES = new Set([
+          'ai',
+          'db',
+          'fs',
+          'tableName',
+          'signalBus',
+          'systemDb', // Getters from base classes
+          '_ai',
+          '_db',
+          '_fs',
+          '_className',
+          '_signalBus',
+          '_tableName', // Private properties
+          '_id',
+          '_slug',
+          '_context',
+          '_loadedRelationships', // SmrtObject private properties
+          'options', // Configuration object
+        ]);
+
+        // Schema properties that should be persisted despite being on base class
+        const SCHEMA_PROPERTIES = new Set([
+          'id',
+          'slug',
+          'context',
+          'name',
+          'created_at',
+          'updated_at',
+        ]);
+
         // First pass: look for Field instances (from field helpers like text())
         for (const key of allProperties) {
           // Skip protected properties - this fixes Issue #13
-          if (key.startsWith('_') || key.startsWith('#') || key === 'options') {
+          if (key.startsWith('_') || key.startsWith('#')) {
+            continue;
+          }
+
+          // Skip base class properties (Issue #75) except whitelisted schema properties
+          if (BASE_CLASS_PROPERTIES.has(key) && !SCHEMA_PROPERTIES.has(key)) {
             continue;
           }
 
@@ -312,11 +349,12 @@ export class ObjectRegistry {
         if (fields.size === 0) {
           for (const key of allProperties) {
             // Skip protected properties
-            if (
-              key.startsWith('_') ||
-              key.startsWith('#') ||
-              key === 'options'
-            ) {
+            if (key.startsWith('_') || key.startsWith('#')) {
+              continue;
+            }
+
+            // Skip base class properties (Issue #75) except whitelisted schema properties
+            if (BASE_CLASS_PROPERTIES.has(key) && !SCHEMA_PROPERTIES.has(key)) {
               continue;
             }
 

--- a/packages/core/src/scanner/ast-scanner.ts
+++ b/packages/core/src/scanner/ast-scanner.ts
@@ -15,6 +15,7 @@ import type {
 export class ASTScanner {
   private program: ts.Program;
   private options: ScanOptions;
+  private baseClassProperties: Set<string> = new Set();
 
   constructor(filePaths: string[], options: ScanOptions = {}) {
     this.options = {
@@ -35,6 +36,9 @@ export class ASTScanner {
       skipLibCheck: true,
       strict: true,
     });
+
+    // Extract properties defined on base classes to exclude from subclass schemas
+    this.baseClassProperties = this.extractBaseClassProperties();
   }
 
   /**
@@ -53,6 +57,40 @@ export class ASTScanner {
     }
 
     return results;
+  }
+
+  /**
+   * Extract properties defined on base classes
+   * These properties should be excluded from subclass schemas to avoid circular serialization
+   */
+  private extractBaseClassProperties(): Set<string> {
+    const baseProperties = new Set<string>();
+
+    for (const sourceFile of this.program.getSourceFiles()) {
+      // Skip declaration files - we only want actual source files
+      if (sourceFile.isDeclarationFile) continue;
+
+      ts.forEachChild(sourceFile, (node) => {
+        if (ts.isClassDeclaration(node)) {
+          const className = node.name?.text;
+
+          // Check if this is one of our base classes
+          if (className && this.options.baseClasses?.includes(className)) {
+            // Extract all property declarations from this base class
+            for (const member of node.members) {
+              if (ts.isPropertyDeclaration(member)) {
+                const propName = this.getPropertyName(member);
+                if (propName) {
+                  baseProperties.add(propName);
+                }
+              }
+            }
+          }
+        }
+      });
+    }
+
+    return baseProperties;
   }
 
   /**
@@ -332,6 +370,18 @@ export class ASTScanner {
     // Skip static properties for now
     if (node.modifiers?.some((m) => m.kind === ts.SyntaxKind.StaticKeyword)) {
       return null;
+    }
+
+    const propertyName = this.getPropertyName(node);
+    if (!propertyName) return null;
+
+    // Skip properties inherited from base classes to avoid circular serialization
+    if (this.baseClassProperties.has(propertyName)) {
+      // Exception: Allow specific SmrtObject schema properties that should be persisted
+      const SCHEMA_PROPERTIES = ['name', 'created_at', 'updated_at'];
+      if (!SCHEMA_PROPERTIES.includes(propertyName)) {
+        return null;
+      }
     }
 
     // Determine field type from initializer or type annotation

